main.go

```diff
@@ -475,22 +475,93 @@ func makeSecretChanges(kubeClient *k8s.Client, secret *corev1.Secret, initiator
 	return status, nil
 }
 
-func processSecret(kubeClient *k8s.Client, secret *corev1.Secret, initiator string) (status string, err error) {
+func isEventExist(kubeClient *k8s.Client, namespace string, name string, event *corev1.Event ) (exist bool,err error){
+	err = kubeClient.Get(context.Background(), namespace, name, event)
+	exist = true
+	if apiErr, ok := err.(*k8s.APIError); ok {
+		if apiErr.Code == http.StatusNotFound {
+			exist = false
+			return exist,nil
+		}
+	}
+	return exist,nil
+}
 
-	status = "failed"
+func postEventAboutStatus(kubeClient *k8s.Client, secret *corev1.Secret, eventType string, action string, reason string, kind string, reportingController string, reportingInstance string)(err error){
+	log.Info().Msgf(" Starting Function")
```

Comment:
Remove the leading space from the message. You can use `Msg` instead of `Msgf` if you don't substitute any variables in the string.

Reply:
I removed this log no needed!